docs/user-guide/advanced_topics/timeouts.md

```diff
@@ -0,0 +1,42 @@
+
+When establishing a network connection sometimes the necessary ports and socket resources are not available in which case the `--network_timeout` applies. This defaults to the `--timeout` but it can be specified independently if desired. The core will use this timeout for retries on network resources before generating an error. The network timeouts are not dependent on the tick timer so will apply even if the timers have been disabled.
+
+## Co-simulation operation
+
+While a cosimulation is executing, bad things can happen networks can go down, federates can crash. If left on its own this may result in a co-simulation hanging and doing nothing waiting for the failed federate. Detecting and handling this is the primary role of the heartbeat timer. If a broker has not received any communication in a tick, it sends out a ping to its parent or children. IF the parent responds then no action is taken. If no response is received in a certain period of time an error is generated and the process terminated. Sometime cores are not able to respond to pings in which case the `--slowresponding` flag should be used on that federate/core/broker to prevent co-sim termination from lack of ping responses. The `--debugging` option specified early is really a macro flag which turns on `--slowresponding` and `--disable_timer`. The primary use case is for doing step wise debugging of a federate in a code debugger, otherwise the ticks and other federates could cause issues from the timeouts and pings to the federate being debugged.
+
+At the federate level a --granttimeout option is available. This depends on ASIO being compiled in and works similar to the real time mode of federates. At present it is primarily for diagnostics as it cannot generate an error, though this may change at some point in the future. It is disabled by default since some federates could legitimately take a long time to execute so having a timeout on others doesn't make sense. So any grant timeout needs to be handled based on the user knowledge of how the different federates operate. This operation is subject to change but at present the operation is handled in 4 stages.
```

Comment:
```suggestion
At the federate level a `--granttimeout` option is available. This depends on ASIO being compiled in and works similar to the real time mode of federates. At present it is primarily for diagnostics as it cannot generate an error, though this may change at some point in the future. It is disabled by default since some federates could legitimately take a long time to execute so having a timeout on others doesn't make sense. So any grant timeout needs to be handled based on the user knowledge of how the different federates operate. This operation is subject to change but at present the operation is handled in 4 stages.
```